### DISCUSSION TITLE
bin/docker-compose: line 28: docker-compose: command not found

### DISCUSSION THREAD

**Discussion by lamba** [markshust/docker-magento#1054] [Q&A] [answered]
I had an old (corrupt) instance of Docker on my Mac, which I uninstalled. Reinstalled Docker using "brew install --cask docker". Ran the Docker app and made sure it is in "running" state. Then ran the curl command to install docker-magento and I get the error in the subject line. Any suggestions?

**Comment by YevhenZvieriev** [comment, marked as answer]
Hi, @lamba 

This is because Docker Desktop (Mac/Win) or Docker Compose (Linux) is not installed. Once you install that prerequisite, it should work.

**Reply by lamba** [reply]
Thanks. Installing docker-compose helped. Then ran into a credentials issue that was preventing docker-magento from being able to access Docker Desktop, for which the following solution seems to have worked

https://forums.docker.com/t/docker-credential-desktop-exe-executable-file-not-found-in-path-using-wsl2/100225/3

I'm now trying to figure out this

Composer authentication required (repo.magento.com public and private keys):

It actually expecting me to setup the keys on repo.magento.com which has apparently been replaced by Adobe Commerce. I don't see any steps on the setup page about this but let me check again

**Reply by YevhenZvieriev** [reply]
Keys can expire, so you should regenerate them.

It can be useful for you https://github.com/markshust/docker-magento/issues/1028#issuecomment-1868334008

You can copy your public and private keys, and paste them within the one-line script executing.

**Reply by lamba** [reply]
Thanks. I think composer authentication is working now. But even though Docker Desktop is installed and running, I am still getting the following error 

bin/start: line 4: docker: command not found

As a result magento.test is not yet coming up

**Reply by YevhenZvieriev** [reply]
Can you please, verify that it works run: ```sudo docker run hello-world```?

**Reply by lamba** [reply]
I additionally performed the following steps

Added the Docker path to my Bash profile 

export PATH="$PATH:/Applications/Docker.app/Contents/Resources/bin/"

and removed the default Docker container

I saw the Magento container running in Docker Dashboard. However, browsing to https://magento.test doesn't not bring up the site. My guess is that the entry into /etc/hosts did not happen. What entry is supposed to go there?

Hello world also worked but container shows as exited in the DD now

**Reply by YevhenZvieriev** [reply]
Correct! You should add the Docker path to your Bash profile.

I think it is because the one-liner script didn`t finish.

It adds an entry into /etc/hosts during execution, so you don`t need to do it manually.

Do you have a problem when you execute the one-liner script?

**Reply by lamba** [reply]
I'm not seeing any error (see attachment)
<img width="1680" alt="Screenshot 2024-02-03 at 3 26 38 PM" src="https://github.com/markshust/docker-magento/assets/1646112/67fbf066-0318-4d63-ae8b-f7db4919a343">
 
I deleted the project and ran it again from scratch. It's actually not asking for my password to edit /etc/hosts. 

Could the Apache "project directory not empty" be the issue?

<img width="1269" alt="Screenshot 2024-02-03 at 3 30 31 PM" src="https://github.com/markshust/docker-magento/assets/1646112/a9162f61-8d67-4bd5-b2b9-1a425175f3bf">

**Reply by YevhenZvieriev** [reply]
I got it, you have a problem!

You should execute ```bin/removeall```, in the folder of the project, after that, you should remove the folder and recreate it, and after that, try to execute the one-liner again.

**Reply by lamba** [reply]
I did that. It ended with the last line

chmod: cannot access 'bin/magento': No such file or directory

<img width="1662" alt="Screenshot 2024-02-03 at 3 39 32 PM" src="https://github.com/markshust/docker-magento/assets/1646112/1e950fe3-1cd8-420f-9134-0f17d51c6277">

Do I need to manually chmod? I can but I'm thinking it's a symptom of another problem?

**Reply by YevhenZvieriev** [reply]
It is because you got an error  `The URL 'https://repo.magento.com/packages.json' required authentication (HTTP 401)`

It is because of auth.json 

Pay attention to this https://github.com/markshust/docker-magento/issues/1028#issuecomment-1868334008

You should update auth.json file ```nano ~/.composer/auth.json```, just paste the new keys.

```
{
"http-basic": {
        "repo.magento.com": {
                   "username": "<New Public Key>",
                   "password": "<New Private Key>"
              }

}
```
I hope that can help you!

**Reply by lamba** [reply]
Thanks! I think I got rid of that error. But I still have this 

Project directory "/var/www/html/." is not empty.

Is this a problem? https://magento.test is still not coming up.

<img width="1663" alt="Screenshot 2024-02-03 at 4 04 35 PM" src="https://github.com/markshust/docker-magento/assets/1646112/053dafce-970d-4c1b-8a05-c9f8f97157ec">

**Reply by YevhenZvieriev** [reply]
Yes, your error is leading to artifacts being left over from a preexisting install attempt. 

How to fix it:
```
bin/removeall
cd ..
rm -rf yourproject
```
 The ```bin/removeall``` command removes all previous Docker containers & volumes related to the specific project directory you are within. You can then attempt the install process again.

**Reply by lamba** [reply]
Thanks. I thought I tried that earlier. Will try it again. :)

Ok now it's installing properly - I think I had bad keys and that was messing up the install now matter what I did. 

Thanks very much!

**Reply by YevhenZvieriev** [reply]
```Project directory "/var/www/html/." is not empty error```

This message occurs when something fails to execute correctly during an install, and a subsequent install is re-attempted. Unfortunately, when attempting a second (or third) install, the src directory may be no longer empty. This prevents Composer from creating new project because it needs to create new projects within an empty directory.

And you can fix this  by running:
```
bin/removeall
cd ..
rm -rf yourproject
```

**Reply by YevhenZvieriev** [reply]
If you see  

```
"You may now access your Magento instance at https://magento.test"
``` 
After the one-liner completes running it means that you installed Magento 

Congratulations🥳
